src/main/java/com/thed/zephyr/jenkins/model/ZephyrInstance.java

```diff
@@ -117,9 +117,17 @@ public FormValidation doTestConnection(
             String zephyrURL = URLValidator.validateURL(serverAddress);
 
             try {
-                Boolean verifyCredentials = userService.verifyCredentials(serverAddress, username, password);
-                if(!verifyCredentials) {
-                    return FormValidation.error("Username or password is incorrect.");
+                if(upCredentials instanceof UsernamePasswordCredentialsImpl) {
+                    Boolean verifyCredentials = userService.verifyCredentials(serverAddress, ((UsernamePasswordCredentialsImpl) upCredentials).getUsername(),
+                            ((UsernamePasswordCredentialsImpl) upCredentials).getPassword().getPlainText());
+                    if (!verifyCredentials) {
+                        return FormValidation.error("Username or password is incorrect.");
+                    }
+                }else if(upCredentials instanceof StringCredentialsImpl){
+                    Boolean verifyCredentials = userService.verifyCredentials(serverAddress, ((StringCredentialsImpl) upCredentials).getSecret().getPlainText());
+                    if (!verifyCredentials) {
+                        return FormValidation.error("Api token is incorrect.");
+                    }
```

Comment:
Else block should be there to handle any credential other than UsernamePassword and SecretText.